File: frontend/src/api/opportunity.ts
import client from './client'

export interface Opportunity {
  id: string
  source: string
  title: string
  company_name: string
  location_raw: string | null
  country_code: string | null
  remote_option: string | null
  employment_type: string | null
  application_url: string | null
  posted_at: string | null
  is_active: boolean
  created_at: string
}

export interface OpportunityDetail extends Opportunity {
  description: string | null
  requirements: string | null
  salary_min: number | null
  salary_max: number | null
  salary_currency: string | null
}

export interface OpportunityList {
  items: Opportunity[]
  total: number
  page: number
  page_size: number
}

export interface DiscoveryRunResponse {
  new_jobs_found: number
  message: string
}

export async function runDiscovery(): Promise<DiscoveryRunResponse> {
  const { data } = await client.post<DiscoveryRunResponse>('/api/v1/discovery/run')
  return data
}

export async function listOpportunities(params?: {
  page?: number
  page_size?: number
  country_code?: string
  source?: string
}): Promise<OpportunityList> {
  const { data } = await client.get<OpportunityList>('/api/v1/opportunities', { params })
  return data
}

export async function getOpportunity(id: string): Promise<OpportunityDetail> {
  const { data } = await client.get<OpportunityDetail>(`/api/v1/opportunities/${id}`)
  return data
}

File: backend/app/schemas/opportunity.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class OpportunityResponse(BaseModel):
    id: uuid.UUID
    source: str
    title: str
    company_name: str
    location_raw: Optional[str]
    country_code: Optional[str]
    remote_option: Optional[str]
    employment_type: Optional[str]
    application_url: Optional[str]
    posted_at: Optional[datetime]
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class OpportunityDetailResponse(OpportunityResponse):
    description: Optional[str]
    requirements: Optional[str]
    salary_min: Optional[int]
    salary_max: Optional[int]
    salary_currency: Optional[str]


class DiscoveryRunResponse(BaseModel):
    new_jobs_found: int
    message: str


class OpportunityListResponse(BaseModel):
    items: list[OpportunityResponse]
    total: int
    page: int
    page_size: int
